### PR TITLE
Pausable indexer

### PR DESCRIPTION
**Pull request by alicevhenshaw**
## Description
https://github.com/airswap/airswap-protocols/issues/181

Quick description:
- [X] New features

### CODE REVIEW

File: protocols/indexer/contracts/Indexer.sol [hunk 1]
```diff
@@ -213,4 +207,43 @@ contract Indexer is IIndexer, Ownable {
     }
     return new bytes32[](0);
   }
+
+  /**
+    * @notice Allows the owner to pause and unpause the contract
```
Comment (ejwessel) [inline protocols/indexer/contracts/Indexer.sol]: @dev for only callable by owner

File: protocols/indexer/contracts/Indexer.sol [hunk 2]
```diff
@@ -213,4 +207,43 @@ contract Indexer is IIndexer, Ownable {
...
+    */
+  function setPausedStatus(bool _newStatus) external onlyOwner {
+    paused = _newStatus;
+  }
+
+  /**
+    * @notice Allows the owner to unset intent and return tokens
```
Comment (ejwessel) [inline protocols/indexer/contracts/Indexer.sol]: @dev for only callable by owner

File: protocols/indexer/contracts/Indexer.sol (hunk 1 above)
Comment (ejwessel) [reply protocols/indexer/contracts/Indexer.sol]: @param too

File: protocols/indexer/contracts/Indexer.sol (hunk 2 above)
Comment (ejwessel) [reply protocols/indexer/contracts/Indexer.sol]: @param too

File: protocols/indexer/contracts/Indexer.sol [hunk 3]
```diff
@@ -213,4 +207,43 @@ contract Indexer is IIndexer, Ownable {
...
+    */
+  function unsetIntentForUser(address _user, address _signerToken, address _senderToken) external onlyOwner {
+    unsetUserIntent(_user, _signerToken, _senderToken);
+  }
+
+  /**
+    * @notice Internal function that unsets a user's intent and returns tokens
```
Comment (ejwessel) [inline protocols/indexer/contracts/Indexer.sol]: add @param

File: protocols/indexer/contracts/Indexer.sol [hunk 4]
```diff
@@ -54,6 +56,15 @@ contract Indexer is IIndexer, Ownable {
     locatorWhitelist = _locatorWhitelist;
   }
 
+  /**
+    * @notice Modifier to prevent function call unless the contract is not paused
+    *
+    */
+  modifier whenNotPaused() {
```
Comment (dmosites) [inline protocols/indexer/contracts/Indexer.sol]: Suggest `notPaused` to be consistent with `onlyOwner`

File: protocols/indexer/contracts/Indexer.sol [hunk 5]
```diff
@@ -40,6 +40,8 @@ contract Indexer is IIndexer, Ownable {
   // The whitelist contract for checking whether a peer is whitelisted
   address public locatorWhitelist;
 
+  bool public paused = false;
```
Comment (dmosites) [inline protocols/indexer/contracts/Indexer.sol]: Please add a comment above to describe its use

File: protocols/indexer/contracts/Indexer.sol [hunk 6]
```diff
@@ -83,7 +94,7 @@ contract Indexer is IIndexer, Ownable {
     */
   function addToBlacklist(
     address _token
-  ) external onlyOwner {
+  ) external onlyOwner whenNotPaused {
```
Comment (dmosites) [inline protocols/indexer/contracts/Indexer.sol]: Given that the owner can pause and unpause, we can do without the restriction here and leave to `onlyOwner`. Same for `removeFromBlacklist`

Comment (alicevhenshaw) [reply protocols/indexer/contracts/Indexer.sol]: I am inclined to leave the double restriction on `killContract` to help prevent any accidental calls etc. It seems like a function worth protecting

File: protocols/indexer/contracts/Indexer.sol (hunk 4 above)
Comment (alicevhenshaw) [reply protocols/indexer/contracts/Indexer.sol]: This meant I have to rename the `bool paused` to `bool contractPaused` to keep the other modifier consistent.

**Pull request merged**